src/main/java/org/broadinstitute/hellbender/tools/spark/sv/InsertSizeMetricsCollectorSpark.java

```diff
@@ -1,99 +1,256 @@
+        }
+        final SortedMap<GroupInfo, Map<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>>> groupInfo = histogramsAndMetrics.get(i);
+        final Map<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>> groupStats = groupInfo.values().iterator().next();
+        return groupStats;
     }
 
-    public final InsertSizeMetrics getMetrics(){
-        return metrics;
+    /**
+     * Utility getter for retrieving InsertSizeMetrics of a particular group, given its name.
+     * Stats info returned are organized by pair orientations.
+     * If a particular pair orientation is unavailable (i.e. no reads of this group has pairs of that orientation), it is not returned.
+     * @param groupName String representation of the group's name/id, whose InsertSizeMetrics is requested.
+     * @return          the requested group's InsertSizeMetrics information, organized by pair orientations
+     * @throws IllegalArgumentException
+     */
+    public Map<SamPairUtil.PairOrientation, InsertSizeMetrics> getMetrics(final String groupName) throws IllegalArgumentException {
+        final Map<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>> intermediate = getHistogramsAndMetrics(groupName);
+        Map<SamPairUtil.PairOrientation, InsertSizeMetrics> result = new HashMap<>();
+        for(Map.Entry<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>> entry : intermediate.entrySet()){
+            result.put(entry.getKey(), entry.getValue()._2());
+        }
+        return result;
     }
 
-    public final Histogram<Integer> getHitogram(){
-        return htslibHistogram;
+    /**
+     * Utility getter for retrieving InsertSizeMetrics of a particular group, given its name and requested orientation of the pairs.
+     * If no read pairs are available in this particular group of the requested orientation, returns null.
+     * @param groupName   String representation of the group's name/id, whose InsertSizeMetrics information is requested.
+     * @param orientation Requested orientation.
+     * @return            InsertSizeMetrics of the requested group, of the requested orientation (could be null if no reads available)
+     */
+    public InsertSizeMetrics getMetricsOfOrientation(final String groupName, final SamPairUtil.PairOrientation orientation){
+        return getMetrics(groupName).get(orientation);
     }
 
-    // simply return an double array from an RDD of integers for using Apache DescriptiveStatistics
-    private static final double[] getRDDArray(final JavaRDD<Integer> rdd){
-        final List<Integer> tempList = rdd.collect();
-        final double lengthArr[] = new double[tempList.size()];
-        for(int i=0; i<lengthArr.length; ++i){
-            lengthArr[i] = tempList.get(i);
+    /**
+     * Intermediate step worker function where higher level (library, sample, all_reads) hand-rolled histograms are
```

Comment:
Try to describe this method in terms of its inputs and outputs rather than how it fits in the control flow of your tool. For example: "This method takes a list of read group-level histograms and returns a list of histograms that have been aggregated at the sample or library level... etc."